### ISSUE TITLE
[Bug] 3.24.2 - up to - 3.25.3: `href` empty from links in rendering tests

### ISSUE THREAD

**Issue by esbanarango** [buschtoens/ember-link#554]
### 🔬 Minimal Reproduction
Describe steps to reproduce. If possible, please, share a link with a minimal reproduction.

### 😕 Actual Behavior
A clear and concise description of what is happening.

### 🌍 Environment

- Ember: -
- Node.js/npm: -
- OS: -
- Browser: -

**Comment by rwjblue** [comment]
Depending on how this addon does URL generation, I suspect it has to do roughly the same thing as the Ember PR's:

* https://github.com/emberjs/ember.js/pull/19080
* https://github.com/emberjs/ember.js/pull/19326
* https://github.com/emberjs/ember.js/pull/19387
* https://github.com/emberjs/ember.js/pull/19395

**Comment by Turbo87** [comment]
@esbanarango that is roughly how Ember up until now also behaved and the addon tries to follow what Ember itself does. we even explicitly test for this behavior in the test suite here.

**Comment by esbanarango** [comment]
@Turbo87 mmmm that's weird. Looks like this is actually testing it to be "empty" ☹️.

https://github.com/buschtoens/ember-link/blob/master/tests/integration/components/link-test.ts#L40 or to be an ember selector (?) if you're using the `setupLink` hook:

(Using `setupLink` hook).
<img width="632" alt="Garaje_Tests_and_tmp-test_js_—_workspace__Workspace_" src="https://user-images.githubusercontent.com/1156865/110343430-7cf64d00-7ffa-11eb-81e5-dcd6df54f1e0.png">


I don't think this is up with how Ember is behaving now. 

This same test:

```js
module('Integration | Component | TMP', function (hooks) {
  setupRenderingTest(hooks);

  test('href test', async function (assert) {
    assert.expect(1);
    await render(hbs`<LinkTo @route="login">test</LinkTo>`);
    assert.dom('a').hasAttribute('href', `/login`);
  });
});
```

works fine with `ember-source: 3.25.3`. 

But this one breaks with `ember-link: 1.3.1`:

```js
module('Integration | Component | TMP', function (hooks) {
  setupRenderingTest(hooks);

  test('href test', async function (assert) {
    assert.expect(1);
    await render(hbs`
      {{#let (link route="login") as |l|}}
        <a href={{l.url}}>Login</a>
      {{/let}}
    `);
    assert.dom('a').hasAttribute('href', `/login`);
  });
});
```

**Comment by rwjblue** [comment]
Yeah :/ sorry about this. We are trying to make the default behavior appropriate out of the box. From my perspective, the current implemetnation (in 3.24.3 and 3.25.3) is much better...

**Comment by rogeraraujo90** [comment]
I asked if this behavior were a bug in the related issue. Good to know that we will have a fix for it! 👏🏼🧡

**Comment by esbanarango** [comment]
@Turbo87 I tried tackling this issue but I think I missing some context on you guys are matching the way Ember behaves. Is this on your radar 🙏 ?
